src/pages/Dashboard.tsx

```diff
@@ -7,16 +7,35 @@ import { BorderRad, Colors, Shadows } from '../global/styles';
 import { Web3Storage } from 'web3.storage'
 import { WEB3_STORAGE_API_KEY } from '../global/apiKeys';
 import { Button } from 'react-bootstrap';
+import { Contract } from '@ethersproject/contracts'
+import soulMint from '../artifacts/contracts/SoulMint.sol/SoulMint.json'
+import { ethers } from 'ethers'
+import { Spinner } from 'react-bootstrap'
 
 
+export const SouleMintAbi = [
+    "function mintOne(uint256 eventId, address to) public",
+  ];
+
 export function DashBoardPage() {
-const { chainId, account } = useEthers();
+const { chainId, account, library } = useEthers();
 const [selectedFile, setSelectedFile] = useState();
+const [currentIpfsLinks, setCurrentIpfsLinks] = useState<string[]>([]);
+const [isUploading, setIsUploading] = useState<boolean>(false);
+
+const signer = library?.getSigner();
+
+let contract: Contract|null = null;
+const souleMintInterface = new ethers.utils.Interface(SouleMintAbi);
+contract = new Contract('0xf46bB4381BECf28CB0ebf7D6a5127bD04810d0cE', souleMintInterface, signer);
```

Comment:
Probably want to instantiate soulMint for your account instance specifically, from the factory:
https://github.com/d0m0l33/SideQuest/commit/0abdc3629b2eaedef56cf6ee7b24f722c3343e24#diff-ab3e7f0f39bfe2ac287bd1e4321c2f04127698d809f9ac7d33bf55d68dd4d4ecR52